weatherflow2mqtt/weatherflow_mqtt.py

```diff
@@ -166,19 +176,346 @@ async def connect(self) -> None:
             _LOGGER.error("Could not connect to MQTT Server. Error is: %s", e)
             sys.exit(1)
 
+        self.listener = WeatherFlowListener(self.udp_config.host, self.udp_config.port)
+        self.listener.on(
+            EVENT_DEVICE_DISCOVERED, lambda device: self._device_discovered(device)
+        )
         try:
-            self.endpoint = await open_local_endpoint(
-                host=self.udp_config.host, port=self.udp_config.port
-            )
-            _LOGGER.info(
-                "The UDP server is listening on port %s", self.endpoint.address[1]
-            )
+            await self.listener.start_listening()
+            _LOGGER.info("The UDP server is listening on port %s", self.udp_config.port)
         except Exception as e:
             _LOGGER.error(
                 "Could not start listening to the UDP Socket. Error is: %s", e
             )
             sys.exit(1)
 
+        self._queue = asyncio.Queue()
+        self._queue_task = asyncio.ensure_future(self._mqtt_queue_processor())
+
+        fcst_state_topic = MQTT_TOPIC_FORMAT.format(DOMAIN, FORECAST_ENTITY, "state")
+        fcst_attr_topic = MQTT_TOPIC_FORMAT.format(
+            DOMAIN, FORECAST_ENTITY, "attributes"
+        )
+        for sensor in FORECAST_SENSORS:
+            discovery_topic = MQTT_TOPIC_FORMAT.format(DOMAIN, sensor.id, "config")
+            payload: OrderedDict | None = None
+            if self.forecast is not None:
+                payload = self._get_sensor_payload(
+                    sensor, DOMAIN, fcst_state_topic, fcst_attr_topic, VERSION
+                )
+            self._add_to_queue(
+                discovery_topic, json.dumps(payload or {}), qos=1, retain=True
+            )
```

Comment:
The way this is now, it will currently end up under its own unique device since a forecast is not really associated with a sensor. We could, instead, attach it to the hub since the hub is roughly equivalent to a station. But if there are multiple hubs on one network (huge edge case, but still possible), which hub would be appropriate? Or just add it to both in that scenario?

^ Those were my thoughts going through it, which is why I ended up creating it on its own device.

Reply:
I am actually not sure what happens if you have more than 1 Hub. They both broadcast on the same port, so which one would be picked up? I have 2 Hubs available (Only 1 operational right now) so I will test it as soon as I have time to set it up.

But yes, attach the forecast to the Hub, that makes most sense.

Reply:
Just fired up a container with the changes, and I am not getting any forecast data, so for some reason, which I cannot figure out, it does not seem to fire the event, despite the Vars being set in the Docker command.

Reply:
CORRECTION: After a few Restarts the Forecast data is now populating. So no error here.

Reply:
Final Comment here: Could you add a log entry so that people can see that the Forecast sensor is created. I guess that was part of why I missed that it was there. Currently it only shows for the other sensors.

Reply:
> They both broadcast on the same port, so which one would be picked up?

Both hubs _should_ be picked up in that scenario. While they both broadcast via the same port, they have a unique IP address which allows them to operate independently. There can only be one listener attached to an address/port combo however, which is the "0.0.0.0"/50222 address/port (essentially a non-specific address so it can just listen for these datagrams that are announced on the network via that port).

Reply:
> Final Comment here: Could you add a log entry so that people can see that the Forecast sensor is created. I guess that was part of why I missed that it was there. Currently it only shows for the other sensors.

Done